kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -277,29 +277,29 @@ private[kyo] object IOPromise extends IOPromisePlatformSpecific:
                     f(panic)
                     self
                 def waiters: Int = self.waiters + 1
-                def run(v: Result[E, A]) =
+                def run[E2 >: E, A2 >: A](v: Result[E2, A2]) =
                     self
 
-        final def merge(tail: Pending[E, A]): Pending[E, A] =
+        final def merge[E2 >: E, A2 >: A](tail: Pending[E2, A2]): Pending[E2, A2] =
 
-            @tailrec def runLoop(p: Pending[E, A], v: Result[E, A]): Pending[E, A] =
+            @tailrec def runLoop[E3 >: E2, A3 >: A2](p: Pending[? <: E3, ? <: A3], v: Result[E3, A3]): Pending[E3, A3] =
                 p match
                     case _ if (p eq Pending.Empty) => tail
-                    case p: Pending[E, A]          => runLoop(p.run(v), v)
+                    case p: Pending[?, ?]          => runLoop(p.run(v), v)
 
-            @tailrec def interruptLoop(p: Pending[E, A], panic: Panic): Pending[E, A] =
+            @tailrec def interruptLoop(p: Pending[E, A], panic: Panic): Pending[E2, A2] =
                 p match
                     case _ if (p eq Pending.Empty) => tail
                     case p: Pending[E, A]          => interruptLoop(p.interrupt(panic), panic)
 
-            new Pending[E, A]:
-                def waiters: Int            = self.waiters + 1
-                def interrupt(panic: Panic) = interruptLoop(self, panic)
-                def run(v: Result[E, A])    = runLoop(self, v)
+            new Pending[E2, A2]:
+                def waiters: Int                               = self.waiters + tail.waiters
```

Comment:
The method was incorrectly not considering the number of waiters in the tail. The waiter count is only used by `toString`, though.